### ISSUE TITLE
ProjectionInfo.ToProj4String() returns differs for the same projectio

### ISSUE THREAD

**Issue by pedwik** [DotSpatial/DotSpatial#875]
I was about to add this test to ProjectionInfoTests.cs but it fails:
```
 /// <summary>
        /// Proj4 string esri comparison test for SWEREF99TM.
        /// </summary>
        [Test]
        public void Proj4EsriComparisonTest_SWEREF99TM()
        {
            var infoBuiltIn = KnownCoordinateSystems.Projected.NationalGridsSweden.SWEREF99TM;

            // ESRI string retrieved from http://spatialreference.org/ref/epsg/3006/esriwkt/
            const string esri = "PROJCS[\"SWEREF99 TM\",GEOGCS[\"SWEREF99\",DATUM[\"SWEREF99\",SPHEROID[\"GRS 1980\",6378137,298.257222101,AUTHORITY[\"EPSG\",\"7019\"]],TOWGS84[0,0,0,0,0,0,0],AUTHORITY[\"EPSG\",\"6619\"]],PRIMEM[\"Greenwich\",0,AUTHORITY[\"EPSG\",\"8901\"]],UNIT[\"degree\",0.01745329251994328,AUTHORITY[\"EPSG\",\"9122\"]],AUTHORITY[\"EPSG\",\"4619\"]],UNIT[\"metre\",1,AUTHORITY[\"EPSG\",\"9001\"]],PROJECTION[\"Transverse_Mercator\"],PARAMETER[\"latitude_of_origin\",0],PARAMETER[\"central_meridian\",15],PARAMETER[\"scale_factor\",0.9996],PARAMETER[\"false_easting\",500000],PARAMETER[\"false_northing\",0],AUTHORITY[\"EPSG\",\"3006\"],AXIS[\"y\",EAST],AXIS[\"x\",NORTH]]";
            var infoFromEsri = ProjectionInfo.FromEsriString(esri);

            var expected = infoFromEsri.ToProj4String();
            var actual = infoBuiltIn.ToProj4String();

            Assert.AreEqual(expected, actual);
        }
```

The obtained test values are:
```
expected = " +x_0=500000 +y_0=0 +k_0=0.9996 +lat_0=0 +lon_0=15 +proj=tmerc +a=6378137 +b=6356752.31414036 +no_defs"
actual = " +x_0=500000 +y_0=0 +k_0=0.9996 +lon_0=14.999999999999982 +zone=33 +proj=utm +towgs84=0,0,0 +ellps=GRS80 +no_defs"
````
Observations:
*The CentralMeridian (lon_0) property differs between the two with a small amount, the question is if not such small differences should be tolerated? 
*The built in projection has LatitudeOfOrigin = null, while the expected has LatitudeOfOrigin = 0 (lat_0). Should the ProjectionInfo.ToProj4String()-method as default translate null to 0 in this case?
*The projection created from the esri string has a UTM zone attribute (33) and a different value for the proj attribute. Is proj=tmerc equivalent to proj = utm zone 33?

**Comment by mogikanin** [comment]
@pedwik did you find something on it?

**Comment by pedwik** [comment]
I am no expert in projections, but have asked some people who are and am waiting for their response on the rounding. However, it seems to me that rounding to the 13th decimal for the "Longitude of origin"-parameter should be safe. 

The UTM code represents a Transverse Mercatur (tmerc) projection with a predefined set of parameters. The testing for equivalency of two Proj4-strings for two equivalent projections, one with the implicit utm code as argument, and the other with the explicit tmerc-arguments, will be false negative. 

It seems a bad idea to use Proj4-string and the Esri-string for comparing two projections. It would be better to utilize the ProjectionInfo-parameters for comparison, don't you think?

**Comment by mogikanin** [comment]
> It seems a bad idea to use Proj4-string and the Esri-string for comparing two projections. It would be better to utilize the ProjectionInfo-parameters for comparison, don't you think?

True, and your test proves that. Comparing 2 strings is much simpler than comparing a lot of of ProjectionInfo parameters. This is why it was done in this way.
This [fork](https://github.com/h0st1le/DotSpatial/commit/fab5f47f8459ec0ff23f78a5087f94eb84bf024e) should solve such errors. Only need to check twice that nothing missed.

**Comment by jany-tenaj** [comment]
@mogikanin is there any reason not to integrate the fix from the fork into ours?

**Comment by mogikanin** [comment]
I think we should ask @h0st1le, fork author, provide pull request with that changes. Also unit-tests are necessary.
